lib/streamlit/elements/widgets/slider.py

```diff
@@ -166,14 +183,195 @@ def serialize(self, v: Any) -> list[Any]:
 
 
 class SliderMixin:
-    @gather_metrics("slider")
+    # For easier readability, all the arguments with un-changing types across these overload signatures have been
+    # collapsed onto a single line.
+
+    # fmt: off
+    # HANDLE NUMERIC TYPES
+
+    # if "value: SliderNumericT", "min/max: SliderNumericT | None", and step is the same type or not provided,
+    # then we return a SliderNumericT.
+    @overload
+    def slider(
+        self,
+        label: str,
+        min_value: SliderNumericT | None,
+        max_value: SliderNumericT | None,
+        value: SliderNumericT,
+        step: StepNumericT[SliderNumericT] | None = None,
+        format: str | None = None, key: Key | None = None, help: str | None = None, on_change: WidgetCallback | None = None, args: WidgetArgs | None = None, kwargs: WidgetKwargs | None = None, *, disabled: bool = False, label_visibility: LabelVisibility = "visible"
+    ) -> SliderNumericT:
+        ...
+
+    # if "value: SliderNumericT", and all the other numerical arguments are the same type or not provided,
```

Comment:
nit: should this be `value=SliderNumericT` to emphasize we're passing by kwarg here? (similar comments in a few other places below)

Reply:
Sure, can notate it that way.